### PR TITLE
Integrates adjudication of questions on L2 via the zkEVM bridge

### PR DESCRIPTION
**Pull request by edmundedgar**
feature-fork-integration with the history simplified

### CODE REVIEW

File: contracts/MoneyBox.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-3.0-only
```
Comment (josojo) [inline contracts/MoneyBox.sol:1]: Can we push all the MoneyBox contracts into the mixin folder? I think they are just "utility for us" and don't belong to the core.

File: contracts/MoneyBoxUser.sol [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+pragma solidity ^0.8.20;
+
+import {MoneyBox} from "./MoneyBox.sol";
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+
+abstract contract MoneyBoxUser {
+
+    function _calculateMoneyBoxAddress(address _creator, bytes32 _salt, address _token) internal pure returns (address) {
+
+        return Create2.computeAddress(
```
Comment (josojo) [inline contracts/MoneyBoxUser.sol:12]: Why can't we call Create2.computeAddress direclty in the other contract? What is the MoneyBoxUser wrapper for?

**Review by josojo — APPROVED** [review]
Two nits... then its ready to merge

File: contracts/MoneyBoxUser.sol (hunk 2 above)
Comment (edmundedgar) [reply contracts/MoneyBoxUser.sol:12]: ok, can do it that way

Comment (edmundedgar) [reply contracts/MoneyBoxUser.sol:12]: on second thought looking at the changes the fact that you have to hash some stuff together is an implementation detail you can hide, taking it out will make the code a bit noisier. So mildly opposed to doing it, but OK either way.

File: contracts/MoneyBox.sol (hunk 1 above)
Comment (edmundedgar) [reply contracts/MoneyBox.sol]: fixed in [0556a3a](https://github.com/RealityETH/subjectivocracy/pull/98/commits/0556a3a1d09ebd3af8195dff81bc5e77fbd6f9cf)

File: contracts/MoneyBoxUser.sol (hunk 2 above)
Comment (josojo) [reply contracts/MoneyBoxUser.sol]: Okay, Should we then make it a library contract, instead of an abstract contract?

Comment (edmundedgar) [reply contracts/MoneyBoxUser.sol]: I may be wrong but I thought a library would be deployed as another contract, which is more gas and more complicated deployment. We don't want that, we just want to add a feature to the contract that calls it.

Comment (edmundedgar) [reply contracts/MoneyBoxUser.sol]: OK, for consistency I'll name/format it like CreateChildren.createChildren, something like `CalculateMoneyBoxAddress._calculateMoneyBoxAddress`

Comment (edmundedgar) [reply contracts/MoneyBoxUser.sol]: Done in [1ebd8bc](https://github.com/RealityETH/subjectivocracy/pull/98/commits/1ebd8bc291cd2efbc98025ad0853e1e5663f50ae)

**Pull request merged**